### DISCUSSION TITLE
Annotations, how to create them?

### DISCUSSION THREAD

**Discussion by sblantipodi** [community/community#26703] [Actions]
<p>As title.<br>
Is it possible to create annotations? If yes, how?</p>
<p>Thanks</p>

**Comment by BrightRan** [comment, marked as answer]
<p>You can use the following <strong><a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions" rel="nofollow noopener">Workflow commands</a></strong> to create <strong>W</strong>** arning <strong> and</strong> Error** annotations in the workflow:</p>
<p>◆ <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message" rel="nofollow noopener"><strong>Setting a warning message</strong></a></p>
<p>for example:</p>
<pre><code>- name: Generate Warning Annotations
  run: |
    text="This is a Warning"
    echo "::warning file=app.js,line=1,col=5::$text"
</code></pre>
<p>◆ <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message" rel="nofollow noopener"><strong>Setting an error message</strong></a></p>
<p>for example:</p>
<pre><code>- name: Generate Error Annotations
  run: |
    text="This is an Error"
    echo "::error file=app.js,line=10,col=15::$text"
</code></pre>
<p><img src="https://user-images.githubusercontent.com/46307996/181099304-a6ab1d98-a6a8-4d09-ac15-cd1532e09d53.png" alt="Annotations.PNG" data-base62-sha1="dU6aVSl07yYsPZu6vOtm4nFE9ye" width="356" height="319"></p>
<p><strong>NOTE:</strong>  currently, GitHub Actions does not support too much annotations in a workflow run, the below is a summary of the limitation:</p>
<ul>
<li>10 warning annotations and 10 error annotations for per step</li>
<li>50 annotations for per job (sum of annotations from all the steps)</li>
<li>50 annotations for per workflow run (separate from the job annotations, these annotations aren’t created by users)</li>
</ul>

**Comment by sblantipodi** [comment]
<p>thanks for the reply.</p>
<p>is there a simple INFO annotations?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sblantipodi">@sblantipodi</a> ,</p>
<blockquote>
<p>is there a simple INFO annotations?</p>
</blockquote>
<p>Sorry, what do you mean about “a simple INFO annotations”? I’m not very clear your meaning.</p>

**Comment by sblantipodi** [comment]
<p>it’s not an error, it’s not a warning, it’s a simple message that says:</p>
<p>“Hey, COOL, your build is under 400KB.”</p>
<p>It’s an informative message, no warning, no errors.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sblantipodi">@sblantipodi</a> ,</p>
<p>Currently, we seems only can set  <strong>Error</strong> or <strong>Warning</strong> messages to display on the <strong>Annotations</strong> section in a workflow run.</p>
<p>Actually I’m not sure if it supports other types messages, such as such as <strong>Information</strong> or <strong>Notice</strong>.</p>
<p>I have reported this question to the appropriate engineering team for further discussion. If they have any response, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>

**Comment by sblantipodi** [comment]
<p>love this team! thanks</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sblantipodi">@sblantipodi</a> ,</p>
<p>Sorry for late response.</p>
<p>According to the reply from the appropriate engineering team, currently only  <strong>Error</strong>  and  <strong>Warning</strong>  are supported on the  <strong>Annotations</strong>.</p>
<p>Maybe, a as workaround, you can consider using  <strong>Warning</strong>  to show the information you want.</p>
<p>If your projects really need more message types on <strong>Annotations</strong> , I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Reply by Tinabates** [reply]
I’m concerned about someone building withjson files npm trying to steal my GitHub files and tokens from solana if there is anything you can do to prevent this it would help 12/10/24 I did set 2fa

**Reply by Tinabates** [reply]
He was building solana spl build and I think he switch my setting to he is administrator his name is Mike Hiendreichs gos by bugs_private key

**Comment by sblantipodi** [comment]
<p>I’m so impressed for the service, I really appreciate you answer.<br>
Thanks!</p>

**Comment by ptorngren** [comment]
<p>For any googlers arriving here (like me):<br>
As of July 2021, there is a “<strong>notice</strong>” annotation (see <a href="https://github.com/actions/toolkit" rel="noopener nofollow ugc">GitHub Actions Toolkit</a>):<br>
<code>echo "::notice::my message text"</code></p>

**Reply by Tofandel** [reply]
Does the notice support multiline or markdown?

**Reply by RafaelBNN** [reply]
I don't know about markdown, but you can force a multiline annotation using `%A0`: https://github.com/orgs/community/discussions/25176#discussioncomment-3246718

**Comment by otter-computer** [comment]
<p>Thanks for updating this thread with new info, future searchers appreciate it :smiling_face_with_three_hearts:</p>

**Comment by nichiporenko** [comment]
Thanks! Is there any way to get the text of annotation from the previously finished step?

**Reply by ucyo** [reply]
You can pass information from a preceding job to a succeeding job using [job outputs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#overview). The final step with annotation can then access all this outputs

**Comment by matteocontrini** [comment]
An alternative which may be more appropriate in some cases is to use Markdown-based job summaries, as documented here: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#adding-a-job-summary
